### PR TITLE
Remove duplicate paths

### PR DESCRIPTION
**Pull request by KTamas-Bitrise**

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -136,7 +137,7 @@ func NewCacheModel(configs *ConfigsModel) *CacheModel {
 
 	return &CacheModel{
 		PathList: splittedPaths,
-		//FilePathMap:         map[string]string{},
+		//FilePathMap:       map[string]string{},
```
Comment (trapacska) [inline main.go]: Please do not uncomment this line, can be used anywhere else, so this way we won't run into a nil map

File: main.go [hunk 2]
```diff
@@ -454,11 +455,33 @@ func (cacheModel *CacheModel) CompareFilePathMaps(currentFilePathsMap map[string
...
 }
 
+func cleanDuplicatePaths(paths []string) []string {
+	cleanedPaths := []string{}
+	for _, item := range paths {
+		if item == "" {
+			continue
+		}
+		cleanPath := path.Clean(item)
+		cleanedPaths = append(cleanedPaths, cleanPath)
```
Comment (trapacska) [inline main.go]: Please use: https://github.com/bitrise-io/go-utils/blob/master/sliceutil/sliceutil.go#L30 for checking if `cleanPath` is already in the `cleanedPaths` slice. This way the following lines is not required.

File: main.go [hunk 3]
```diff
@@ -537,9 +560,9 @@ func (cacheModel *CacheModel) CleanPaths() error {
 		}
 	}
 	cacheModel.PathList = cleanedPathList
-
 	cleanedIgnoredPathList := []string{}
-	for _, path := range cacheModel.IgnoreList {
+	ignoreListWithoutDuplicates := cleanDuplicatePaths(cacheModel.IgnoreList)
```
Comment (trapacska) [inline main.go]: Please do not filter ignore patterns, it is not required as it is not an issue if one pattern is checked multiple times for a path.

File: main.go (hunk 1 above)
Comment (KTamas-Bitrise) [reply main.go]: nem uncommenteltem, ket spacet vettem ki

Comment (viktorbenei) [reply main.go]: this is a public repo, please use english (in private ones too ;) )

File: main.go (hunk 2 above)
Comment (KTamas-Bitrise) [reply main.go]: done

File: main.go (hunk 3 above)
Comment (KTamas-Bitrise) [reply main.go]: done

File: main.go (hunk 1 above)
Comment (trapacska) [reply main.go]: ops, sorry

Comment (KTamas-Bitrise) [reply main.go]: sry

**Pull request merged**